robots.js

```diff
@@ -0,0 +1,69 @@
+// Global variables to hold list of all fighters, and user-selected fighter 1 and 2
+let fighters = {};
+let fighter1 = {};
+let fighter2 = {};
+
+// XML call to load fighters
+var data = $.getJSON("characters.json");
+data.then(Main).catch(console.error);
+
+function Main() {
+  // Original Robot prototype
+  let Robot = function() {
+    this.name = "";
+  };
+
+  // Adding properties to Robot prototype for determining health and damage
+  Robot.prototype.setMaxHealth = function(maxHealth,minHealth){
+    this.maxHealth = Math.ceil(Math.random() * (maxHealth - minHealth) + minHealth) || 0;
+  };
+
+  Robot.prototype.setMaxDamage = function(maxDamage,minDamage){
+    this.maxDamage = Math.ceil(Math.random() * (maxDamage - minDamage) + minDamage) || 0;
+  };
+
+  // Declaring robot types and generating prototypes
+  let Transformer = function(){};
+  let Ranger = function(){};
+  let Reboot = function(){};
+
+  Transformer.prototype = new Robot();
+  Transformer.prototype.type = "transformer";
+  Ranger.prototype = new Robot();
+  Ranger.prototype.type = "ranger";
+  Reboot.prototype = new Robot();
+  Reboot.prototype.type = "reboot";
+
```

Comment:
These Types don't really have any unique properties, so the prototype chain is mostly wasted on them. No need to go back and add stuff here, but just recognize that in the future it's balancing unique and shared properties that drives the creation of objects on a prototype chain, not just meeting the requirement to create an object that inherits from Robot.